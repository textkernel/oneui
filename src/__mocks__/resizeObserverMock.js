export class ResizeObserverMock {
    observe = jest.fn();

    unobserve = jest.fn();

    disconnect = jest.fn();
}
