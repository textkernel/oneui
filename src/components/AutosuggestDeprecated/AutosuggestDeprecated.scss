.AutosuggestDeprecated {
    position: relative;
    min-height: 38px;

    &__main {
        position: relative;
        background-color: var(--color-background);
        transition-property: box-shadow, transform, border-color;
        transition-duration: var(--transition-duration);
        border-radius: var(--border-radius);
        z-index: 400;

        &--focused {
            position: absolute;
            width: 100%;
            z-index: 450;
        }

        &--isProminent {
            transform-origin: left top;
        }
    }

    &__field {
        padding: 0;
    }

    &__wrapper {
        cursor: pointer;
        display: flex;
        flex-wrap: wrap;
        align-items: center;
        min-height: 30px;
        height: 30px; // make IE happier
        outline: none;
        padding: var(--spacing-normal);

        &--focused {
            height: auto;
        }
    }

    &__spacedElem {
        margin: var(--spacing-normal);
    }

    &__input {
        border: 0;
        margin: var(--spacing-normal);
        flex: 1;
        font-size: var(--font-size-normal);
        min-width: 150px;
        background-color: var(--color-background);
        color: var(--color-acent);

        &:focus {
            outline: none;
        }

        &::placeholder {
            color: var(--color-neutral-25);
            opacity: 1;
        }
    }

    &__list--focused {
        max-height: 214px;
        overflow: auto;
    }

    &__loaderContainer {
        width: 250px;
    }
}
