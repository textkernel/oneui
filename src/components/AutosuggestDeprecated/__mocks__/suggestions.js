export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
