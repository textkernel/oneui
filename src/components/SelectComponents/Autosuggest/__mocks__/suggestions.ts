export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export type SuggestionsType = (typeof SUGGESTIONS)[number];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'star' },
    { name: 'Moon', type: 'star' },
    { name: 'Mercury', type: 'planet' },
    { name: 'Venus', type: 'planet' },
    { name: 'Earth', type: 'planet' },
    { name: 'Mars', type: 'planet' },
    { name: 'Jupiter', type: 'planet' },
    { name: 'Saturn', type: 'planet' },
    { name: 'Neptune', type: 'planet' },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
