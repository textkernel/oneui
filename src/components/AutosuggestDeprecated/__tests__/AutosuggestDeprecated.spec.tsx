import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { AutosuggestDeprecated } from '../AutosuggestDeprecated';
import { SUGGESTION_TO_STRING, SuggestionsType } from '../__mocks__/suggestions';

describe('AutosuggestDeprecated', () => {
    const suggestionToString = SUGGESTION_TO_STRING;
    const selectedPlaceholder = '';
    const inputPlaceholder = 'type here...';
    const defaultInputValue = 'default value';
    const noSuggestionsPlaceholder = 'No suggestions...';
    const clearTitle = 'Clear';
    const mockOnSelectionChange = jest.fn();
    const mockOnInputValueChange = jest.fn();
    const mockOnClearAllSelected = jest.fn();
    const mockOnBlur = jest.fn();

    let suggestionsList: SuggestionsType[] = [];
    const getSuggestions = () => suggestionsList;
    let selectedSuggestions: SuggestionsType[] = [];

    let view: RenderResult;

    const setFocusOnInput = async (user) => {
        await user.click(screen.getByRole('textbox'));
    };

    beforeEach(() => {
        view = render(
            <AutosuggestDeprecated
                selectedSuggestions={selectedSuggestions}
                getSuggestions={getSuggestions}
                suggestionToString={suggestionToString}
                selectedPlaceholder={selectedPlaceholder}
                inputPlaceholder={inputPlaceholder}
                clearTitle={clearTitle}
                noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                onSelectionChange={mockOnSelectionChange}
                onInputValueChange={mockOnInputValueChange}
                onBlur={mockOnBlur}
                onClearAllSelected={mockOnClearAllSelected}
                showClearButton
            />
        );
    });

    afterEach(() => {
        suggestionsList = [];
        selectedSuggestions = [];
    });

    describe('rendering', () => {
        it('should initially render empty component correctly', () => {
            expect(view.container).toMatchSnapshot();
        });

        it('should render a component with default value', () => {
            view = render(
                <AutosuggestDeprecated
                    selectedSuggestions={selectedSuggestions}
                    getSuggestions={getSuggestions}
                    suggestionToString={suggestionToString}
                    selectedPlaceholder={selectedPlaceholder}
                    inputPlaceholder={inputPlaceholder}
                    defaultInputValue={defaultInputValue}
                    clearTitle={clearTitle}
                    noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                    onSelectionChange={mockOnSelectionChange}
                    onInputValueChange={mockOnInputValueChange}
                    onBlur={mockOnBlur}
                    onClearAllSelected={mockOnClearAllSelected}
                    showClearButton
                />
            );
            expect(screen.getAllByRole('textbox')[1]).toHaveAttribute('value', defaultInputValue);
        });

        it('should render empty component correctly when focused', async () => {
            const user = userEvent.setup();
            await setFocusOnInput(user);
            expect(screen.getByRole('combobox')).toHaveClass(
                'AutosuggestDeprecated__main AutosuggestDeprecated__main--focused'
            );
            expect(screen.getByRole('searchbox')).toHaveClass(
                'AutosuggestDeprecated__wrapper AutosuggestDeprecated__wrapper--focused'
            );
        });
    });
});
