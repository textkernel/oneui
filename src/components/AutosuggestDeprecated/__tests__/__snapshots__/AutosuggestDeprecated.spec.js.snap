// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutosuggestDeprecated rendering should initially render empty component correctly 1`] = `
<AutosuggestDeprecated
  clearTitle="Clear"
  defaultInputValue=""
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  inputRef={null}
  isFocused={false}
  isLoading={false}
  isMultiselect={false}
  isProminent={false}
  listRef={null}
  listRenderer={null}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  rootRef={null}
  saveSelectedValueToInput={false}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <div
    className="AutosuggestDeprecated"
  >
    <Downshift
      defaultHighlightedIndex={0}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      inputValue=""
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        className="AutosuggestDeprecated__main"
        role="combobox"
      >
        <FieldWrapper
          className="AutosuggestDeprecated__field"
          clearLabel="Clear"
          disabled={false}
          isArrowUp={false}
          isFocused={false}
          onClear={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          showArrow={false}
          showClearButton={false}
        >
          <div
            className="FieldWrapper AutosuggestDeprecated__field"
            disabled={false}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="FieldWrapper__content"
            >
              <div
                className="AutosuggestDeprecated__wrapper"
                role="searchbox"
                tabIndex="0"
              >
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-labelledby="downshift-0-label"
                  autoComplete="off"
                  className="AutosuggestDeprecated__input"
                  data-lpignore={true}
                  id="downshift-0-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="type here..."
                  value=""
                />
              </div>
              <List
                aria-labelledby="downshift-0-label"
                doSelectOnNavigate={false}
                id="downshift-0-menu"
                isControlledNavigation={true}
                isDivided={false}
                role="listbox"
              >
                <ul
                  aria-labelledby="downshift-0-label"
                  className="List"
                  id="downshift-0-menu"
                  role="listbox"
                />
              </List>
            </div>
          </div>
        </FieldWrapper>
      </div>
    </Downshift>
  </div>
</AutosuggestDeprecated>
`;

exports[`AutosuggestDeprecated rendering should render all suggestions from the list 1`] = `
<AutosuggestDeprecated
  clearTitle="Clear"
  defaultInputValue=""
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  inputRef={null}
  isFocused={false}
  isLoading={false}
  isMultiselect={false}
  isProminent={false}
  listRef={null}
  listRenderer={null}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  rootRef={null}
  saveSelectedValueToInput={false}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <div
    className="AutosuggestDeprecated"
  >
    <Downshift
      defaultHighlightedIndex={0}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      inputValue=""
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={true}
        aria-haspopup="listbox"
        aria-labelledby="downshift-5-label"
        aria-owns="downshift-5-menu"
        className="AutosuggestDeprecated__main AutosuggestDeprecated__main--focused"
        role="combobox"
      >
        <FieldWrapper
          className="AutosuggestDeprecated__field"
          clearLabel="Clear"
          disabled={false}
          isArrowUp={false}
          isFocused={true}
          onClear={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          showArrow={false}
          showClearButton={false}
        >
          <div
            className="FieldWrapper FieldWrapper--isFocused AutosuggestDeprecated__field"
            disabled={false}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="FieldWrapper__content"
            >
              <div
                className="AutosuggestDeprecated__wrapper AutosuggestDeprecated__wrapper--focused"
                role="searchbox"
                tabIndex="0"
              >
                <input
                  aria-activedescendant="downshift-5-item-0"
                  aria-autocomplete="list"
                  aria-controls="downshift-5-menu"
                  aria-labelledby="downshift-5-label"
                  autoComplete="off"
                  className="AutosuggestDeprecated__input"
                  data-lpignore={true}
                  id="downshift-5-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="type here..."
                  value=""
                />
              </div>
              <List
                aria-labelledby="downshift-5-label"
                className="AutosuggestDeprecated__list--focused"
                doSelectOnNavigate={false}
                id="downshift-5-menu"
                isControlledNavigation={true}
                isDivided={false}
                role="listbox"
              >
                <ul
                  aria-labelledby="downshift-5-label"
                  className="List AutosuggestDeprecated__list--focused"
                  id="downshift-5-menu"
                  role="listbox"
                >
                  <ListItem
                    aria-selected={true}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-5-item-0"
                    isHighlighted={true}
                    isSelected={false}
                    key=".$Sun"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={true}
                      className="ListItem ListItem--isHighlighted ListItem--clickable List__item"
                      id="downshift-5-item-0"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Sun
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-5-item-1"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Moon"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-5-item-1"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Moon
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-5-item-2"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Mercury"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-5-item-2"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Mercury
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-5-item-3"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Venus"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-5-item-3"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Venus
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                </ul>
              </List>
            </div>
          </div>
        </FieldWrapper>
      </div>
    </Downshift>
  </div>
</AutosuggestDeprecated>
`;

exports[`AutosuggestDeprecated rendering should render icon correctly if passed 1`] = `
<AutosuggestDeprecated
  clearTitle="Clear"
  defaultInputValue=""
  getSuggestions={[Function]}
  iconNode={
    <IconMatch
      className="test"
    />
  }
  inputPlaceholder="type here..."
  inputRef={null}
  isFocused={false}
  isLoading={false}
  isMultiselect={false}
  isProminent={false}
  listRef={null}
  listRenderer={null}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  rootRef={null}
  saveSelectedValueToInput={false}
  selectedPlaceholder="Current selection"
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <div
    className="AutosuggestDeprecated"
  >
    <Downshift
      defaultHighlightedIndex={0}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      inputValue=""
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-12-label"
        aria-owns={null}
        className="AutosuggestDeprecated__main"
        role="combobox"
      >
        <FieldWrapper
          className="AutosuggestDeprecated__field"
          clearLabel="Clear"
          disabled={false}
          isArrowUp={false}
          isFocused={false}
          onClear={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          showArrow={false}
          showClearButton={false}
        >
          <div
            className="FieldWrapper AutosuggestDeprecated__field"
            disabled={false}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="FieldWrapper__content"
            >
              <div
                className="AutosuggestDeprecated__wrapper"
                role="searchbox"
                tabIndex="0"
              >
                <IconMatch
                  className="AutosuggestDeprecated__spacedElem test"
                >
                  <IconBase
                    className="AutosuggestDeprecated__spacedElem test"
                    preserveAspectRatio={false}
                    viewBox="0 0 107.17 92.73"
                  >
                    <div
                      className="IconBase AutosuggestDeprecated__spacedElem test"
                    >
                      <svg
                        aria-labelledby={null}
                        className="IconBase__svg"
                        role="img"
                        style={
                          Object {
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 107.17 92.73"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0,92.73V80.35A11.25,11.25,0,0,1,11.07,68.91h6.12V63.52c-3.52-2.55-6-7.4-6-12.92,0-8.21,5.21-14.8,11.72-14.8s11.72,6.59,11.72,14.8c0,5.52-2.48,10.37-6,12.92v5.39h6.12A11.25,11.25,0,0,1,45.84,80.35V92.73Z"
                        />
                        <path
                          d="M101.7,0H36.2a5.65,5.65,0,0,0-5.6,5.79v25.3C36.72,34.45,41,41.86,41,50.47a25.62,25.62,0,0,1-.39,4.44H50.91V51.14a3.78,3.78,0,1,1,7.56,0v3.77H78.26V51.14a3.78,3.78,0,1,1,7.55,0v3.77h15.76a5.66,5.66,0,0,0,5.6-5.79V5.79A5.48,5.48,0,0,0,101.7,0ZM81.77,40.11H55.34V35.26a6.5,6.5,0,0,1,6.38-6.59h3.52v-3.1a9,9,0,0,1-3.39-7.4c0-4.71,3-8.61,6.77-8.61s6.77,3.9,6.77,8.61a9,9,0,0,1-3.51,7.4v3.1h3.51a6.5,6.5,0,0,1,6.38,6.59v4.85Z"
                        />
                      </svg>
                    </div>
                  </IconBase>
                </IconMatch>
                <div
                  className="AutosuggestDeprecated__spacedElem"
                >
                  Current selection
                </div>
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-labelledby="downshift-12-label"
                  autoComplete="off"
                  className="AutosuggestDeprecated__input"
                  data-lpignore={true}
                  id="downshift-12-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  value=""
                />
              </div>
              <List
                aria-labelledby="downshift-12-label"
                doSelectOnNavigate={false}
                id="downshift-12-menu"
                isControlledNavigation={true}
                isDivided={false}
                role="listbox"
              >
                <ul
                  aria-labelledby="downshift-12-label"
                  className="List"
                  id="downshift-12-menu"
                  role="listbox"
                />
              </List>
            </div>
          </div>
        </FieldWrapper>
      </div>
    </Downshift>
  </div>
</AutosuggestDeprecated>
`;

exports[`AutosuggestDeprecated rendering should render noSuggestions placeholder when empty suggestions list is passed 1`] = `
<AutosuggestDeprecated
  clearTitle="Clear"
  defaultInputValue=""
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  inputRef={null}
  isFocused={false}
  isLoading={false}
  isMultiselect={false}
  isProminent={false}
  listRef={null}
  listRenderer={null}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "driver",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSelectionChange={[MockFunction]}
  rootRef={null}
  saveSelectedValueToInput={false}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <div
    className="AutosuggestDeprecated"
  >
    <Downshift
      defaultHighlightedIndex={0}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      inputValue="driver"
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={true}
        aria-haspopup="listbox"
        aria-labelledby="downshift-4-label"
        aria-owns="downshift-4-menu"
        className="AutosuggestDeprecated__main AutosuggestDeprecated__main--focused"
        role="combobox"
      >
        <FieldWrapper
          className="AutosuggestDeprecated__field"
          clearLabel="Clear"
          disabled={false}
          isArrowUp={false}
          isFocused={true}
          onClear={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          showArrow={false}
          showClearButton={false}
        >
          <div
            className="FieldWrapper FieldWrapper--isFocused AutosuggestDeprecated__field"
            disabled={false}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="FieldWrapper__content"
            >
              <div
                className="AutosuggestDeprecated__wrapper AutosuggestDeprecated__wrapper--focused"
                role="searchbox"
                tabIndex="0"
              >
                <input
                  aria-activedescendant="downshift-4-item-0"
                  aria-autocomplete="list"
                  aria-controls="downshift-4-menu"
                  aria-labelledby="downshift-4-label"
                  autoComplete="off"
                  className="AutosuggestDeprecated__input"
                  data-lpignore={true}
                  id="downshift-4-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="type here..."
                  value="driver"
                />
              </div>
              <List
                aria-labelledby="downshift-4-label"
                className="AutosuggestDeprecated__list--focused"
                doSelectOnNavigate={false}
                id="downshift-4-menu"
                isControlledNavigation={true}
                isDivided={false}
                role="listbox"
              >
                <ul
                  aria-labelledby="downshift-4-label"
                  className="List AutosuggestDeprecated__list--focused"
                  id="downshift-4-menu"
                  role="listbox"
                >
                  <ListItem
                    className="List__item"
                    disabled={true}
                    highlightContext="default"
                    isHighlighted={false}
                    isSelected={false}
                    key=".0"
                    passDisabledToLi={false}
                  >
                    <li
                      className="ListItem ListItem--disabled List__item"
                    >
                      <div
                        className="ListItem__container"
                        role="presentation"
                      >
                        <Text
                          context="muted"
                          inline={false}
                          key=".0"
                          size="normal"
                        >
                          <p
                            className="Text--context_muted"
                          >
                            No suggestions...
                          </p>
                        </Text>
                      </div>
                    </li>
                  </ListItem>
                </ul>
              </List>
            </div>
          </div>
        </FieldWrapper>
      </div>
    </Downshift>
  </div>
</AutosuggestDeprecated>
`;

exports[`AutosuggestDeprecated rendering should render suggestions also if it passed as an array, and not a function 1`] = `
<AutosuggestDeprecated
  clearTitle="Clear"
  defaultInputValue=""
  getSuggestions={
    Array [
      Object {
        "name": "Sun",
      },
      Object {
        "name": "Moon",
      },
      Object {
        "name": "Mercury",
      },
      Object {
        "name": "Venus",
      },
    ]
  }
  iconNode={null}
  inputPlaceholder="type here..."
  inputRef={null}
  isFocused={false}
  isLoading={false}
  isMultiselect={false}
  isProminent={false}
  listRef={null}
  listRenderer={null}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  rootRef={null}
  saveSelectedValueToInput={false}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <div
    className="AutosuggestDeprecated"
  >
    <Downshift
      defaultHighlightedIndex={0}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      inputValue=""
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={true}
        aria-haspopup="listbox"
        aria-labelledby="downshift-6-label"
        aria-owns="downshift-6-menu"
        className="AutosuggestDeprecated__main AutosuggestDeprecated__main--focused"
        role="combobox"
      >
        <FieldWrapper
          className="AutosuggestDeprecated__field"
          clearLabel="Clear"
          disabled={false}
          isArrowUp={false}
          isFocused={true}
          onClear={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          showArrow={false}
          showClearButton={false}
        >
          <div
            className="FieldWrapper FieldWrapper--isFocused AutosuggestDeprecated__field"
            disabled={false}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="FieldWrapper__content"
            >
              <div
                className="AutosuggestDeprecated__wrapper AutosuggestDeprecated__wrapper--focused"
                role="searchbox"
                tabIndex="0"
              >
                <input
                  aria-activedescendant="downshift-6-item-0"
                  aria-autocomplete="list"
                  aria-controls="downshift-6-menu"
                  aria-labelledby="downshift-6-label"
                  autoComplete="off"
                  className="AutosuggestDeprecated__input"
                  data-lpignore={true}
                  id="downshift-6-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="type here..."
                  value=""
                />
              </div>
              <List
                aria-labelledby="downshift-6-label"
                className="AutosuggestDeprecated__list--focused"
                doSelectOnNavigate={false}
                id="downshift-6-menu"
                isControlledNavigation={true}
                isDivided={false}
                role="listbox"
              >
                <ul
                  aria-labelledby="downshift-6-label"
                  className="List AutosuggestDeprecated__list--focused"
                  id="downshift-6-menu"
                  role="listbox"
                >
                  <ListItem
                    aria-selected={true}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-6-item-0"
                    isHighlighted={true}
                    isSelected={false}
                    key=".$Sun"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={true}
                      className="ListItem ListItem--isHighlighted ListItem--clickable List__item"
                      id="downshift-6-item-0"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Sun
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-6-item-1"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Moon"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-6-item-1"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Moon
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-6-item-2"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Mercury"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-6-item-2"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Mercury
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                  <ListItem
                    aria-selected={false}
                    className="List__item"
                    disabled={false}
                    highlightContext="default"
                    id="downshift-6-item-3"
                    isHighlighted={false}
                    isSelected={false}
                    key=".$Venus"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    passDisabledToLi={false}
                    role="option"
                  >
                    <li
                      aria-selected={false}
                      className="ListItem ListItem--clickable List__item"
                      id="downshift-6-item-3"
                      onMouseDown={[Function]}
                      onMouseMove={[Function]}
                      role="option"
                    >
                      <div
                        className="ListItem__container"
                        onClick={[Function]}
                        role="presentation"
                      >
                        <MarkedText
                          inline={true}
                          key=".0"
                          marker=""
                        >
                          <Text
                            context="default"
                            inline={true}
                            size="normal"
                          >
                            <span>
                              Venus
                            </span>
                          </Text>
                        </MarkedText>
                      </div>
                    </li>
                  </ListItem>
                </ul>
              </List>
            </div>
          </div>
        </FieldWrapper>
      </div>
    </Downshift>
  </div>
</AutosuggestDeprecated>
`;
